apollo-federation/src/sources/connect/query_graph.rs

```diff
@@ -239,7 +230,10 @@ fn process_subselection(
         // Make sure that we have a field on the object type that matches the alias (or the name itself)
         let alias = selection.name();
         let Some(selection_field) = object_type.fields.get(alias) else {
-            todo!("handle error");
+            return Err(FederationError::internal(format!(
+                "expected alias `{alias}` to exist in GraphQL object `{}`",
```

Comment:
aliases are a GraphQL operation thing, not a GraphQL type system thing. you could say "expected field `{alias}` to exist on GraphQL type `{}`"